base/errorshow.jl

```diff
@@ -268,20 +268,25 @@ function showerror(io::IO, ex::MethodError)
             f_is_function = true
         end
         print(io, "no method matching ")
-        show_signature_function(io, isa(f, Type) ? Type{f} : typeof(f))
-        print(io, "(")
+        iob = IOContext(IOBuffer(), io)     # for type abbreviation as in #49795; some, like `convert(T, x)`, should not abbreviate
+        show_signature_function(iob, isa(f, Type) ? Type{f} : typeof(f))
+        print(iob, "(")
         for (i, typ) in enumerate(arg_types_param)
-            print(io, "::", typ)
-            i == length(arg_types_param) || print(io, ", ")
+            print(iob, "::", typ)
+            i == length(arg_types_param) || print(iob, ", ")
         end
         if !isempty(kwargs)
-            print(io, "; ")
+            print(iob, "; ")
             for (i, (k, v)) in enumerate(kwargs)
-                print(io, k, "::", typeof(v))
-                i == length(kwargs)::Int || print(io, ", ")
+                print(iob, k, "::", typeof(v))
+                i == length(kwargs)::Int || print(iob, ", ")
             end
         end
-        print(io, ")")
+        print(iob, ")")
+        str = String(take!(unwrapcontext(iob)[1]))
+        ioc = !haskey(io, :displaysize) ? IOContext(io, :displaysize => displaysize(io)) : io
```

Comment:
So it looks like in this case we will use the default `displaysize()`, but the code path for stacktraces defaults to `typemax`? Is there a good reason for that or should we just change `type_limited_string_from_context` to call `displaysize(io)`?

Reply:
Note this might fully be my fault for putting in typemax instead of using the normal default display size.

Reply:
I wasn't sure why it was `typemax`. I'd certainly support changing it to always call `displaysize(io)`.

Reply:
If we agree to make that change, we can move the call to `displaysize` into `type_limited_string_from_context`.

Reply:
Ok please go ahead and do that.